packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -587,3 +593,43 @@ export class TelemetryNullLogger implements ITelemetryLogger {
     public sendErrorEvent(event: ITelemetryErrorEvent, error?: any): void {}
     public sendPerformanceEvent(event: ITelemetryPerformanceEvent, error?: any): void {}
 }
+
+/**
+ * Take in a event object, stringify any fields that are non-primitives, and return the new event object.
+ * @param event - Event with fields you want to stringify.
+ */
+ function convertToBaseEvent(event: TelemetryEventTypes): TelemetryEventTypes {
+    const newEvent = {
+        ...event
```

Comment:
Not sure you want to do this.  Since `newEvent[key]` is not overwritten if `convertToBasePropertyType` returns `null`, then say there is some arbitrary object on here -- it will stay around.  This is of course how this code used to work, but if we're introducing this conversion layer I think it should be pretty strict.

Best way to approach these corner cases is to try to enumerate them in unit tests. Then you can see for sure the current behavior and be deliberate about the final behavior (this is why they're called `.spec.ts` - they specify the behavior of the code under test)

Reply:
I think we can follow the precedent in the function `[filterValidTelemetryProps](https://github.com/markfields/FluidFramework/blob/89bb5e66a85f67a0e40583676feb922a0d0bdc5e/packages/utils/telemetry-utils/src/errorLogging.ts#L283-L285)`, and log a console error (to catch someone's eye while debugging) and the set a stock value like `"INVALID PROPERTY (arbitrary object)"` (use a different value than that error logging code).

Probably put that code in `convertToBasePropertyType` actually since it's the code that knows why it's invalid.

Reply:
updated `convertToBasePropertyType` to return `"INVALID PROPERTY (arbitrary object)"` at the end of the default case, but did you mean that we should return this in place of `null` (shown below)?

```
            if (!Array.isArray(x)) {
                return null;
            }
```
change to 
```
            if (!Array.isArray(x)) {
                console.error(`UnSupported Format of Logging Property for key ${key}:`, x);
                return "INVALID PROPERTY (arbitrary object)";
            }
```

Reply:
Yeah. And remove the null check on line 607 below.  I'm adding some comments in `convertToBasePropertyType` too, take a look.